### ISSUE TITLE
Copy host registry when read

### ISSUE THREAD

**Issue by blap** [sandboxie-plus/Sandboxie#2650]
### Is your feature request related to a problem or use case?

If you made an alteration in host computer registry but do not want this alteration in a box. Or when you move your portable sandboxie-plus to another computer, maybe this new computer do not have all registry entries of the old computer, causing errors when read. Or fomat one. Is a pain know and copy all registry read by a program in a box to the box.

### Describe the solution you'd like

The idea  is when a sandboxed program read the host registry it will made a copy of this piece of the registry read in the box not only when the program write in registry.

**Comment by DavidXanatos** [comment]
I don't think this is a good idea, the registry entries origining with an other PC will be wrong and may cause even more issues.

**Comment by diversenok** [comment]
As a workaround, you can export the required portion of the registry outside the sandbox and then import it inside. This way, it will persist in the sandbox when modified outside.

**Comment by blap** [comment]
> As a workaround, you can export the required portion of the registry outside the sandbox and then import it inside. This way, it will persist in the sandbox when modified outside and will allow migrating between machines.

Yes. I always do this and add this in INI box: WriteKeyPath=\REGISTRY\*
So the program do not have more access to host registry.
But the initial setup to track what the program read first and copy then to the box got a lot of time.
If I want to fake some IDs from registry to provide more privacy, made then more gereric, It will take a lot more work doing all this manual.

**Comment by DavidXanatos** [comment]
> 
> EDIT: sandboxie already do this with files. So I think it is a good idea to do this with registry too.

this is not the case sandboxie only copys files which are modified into the box, it does not copy files open only for read access.

**Comment by WAZAAAAA0** [comment]
I second the request.

A way to copy every registry key (and file?) that's being READ and not just WRITTEN could be a way to make certain sandboxed closed source apps that need certain keys to be a certain way, "more portable" among different computers.
I understand how niche and unlikely it is to be implemented, but it would still be neat to have a toggle for it.
